### PR TITLE
Skip isDirectory check on known directory paths

### PR DESCRIPTION
**Pull request by tomaz82**

### CODE REVIEW

File: es-core/src/utils/FileSystemUtil.cpp [hunk 1]
```diff
@@ -539,10 +539,10 @@ namespace Utils
 
 //////////////////////////////////////////////////////////////////////////
 
-		std::string removeCommonPath(const std::string& _path, const std::string& _common, bool& _contains)
+		std::string removeCommonPath(const std::string& _path, const std::string& _common, bool& _contains, const bool _skipDirectoryCheck)
 		{
 			const std::string path   = getGenericPath(_path);
-			const std::string common = isDirectory(_common) ? getGenericPath(_common) : getParent(_common);
+			const std::string common = _skipDirectoryCheck ? getGenericPath(_common) : (isDirectory(_common) ? getGenericPath(_common) : getParent(_common));
```
[the remark this answers is not in the document]
Comment (tomaz82) [reply es-core/src/utils/FileSystemUtil.cpp]: Done

File: es-core/src/utils/FileSystemUtil.cpp [hunk 2]
```diff
@@ -486,10 +486,10 @@ namespace Utils
 
 //////////////////////////////////////////////////////////////////////////
 
-		std::string resolveRelativePath(const std::string& _path, const std::string& _relativeTo, const bool _allowHome)
+		std::string resolveRelativePath(const std::string& _path, const std::string& _relativeTo, const bool _allowHome, const bool _skipDirectoryCheck)
 		{
 			const std::string path       = getGenericPath(_path);
-			const std::string relativeTo = isDirectory(_relativeTo) ? getGenericPath(_relativeTo) : getParent(_relativeTo);
+			const std::string relativeTo = _skipDirectoryCheck ? getGenericPath(_relativeTo) : (isDirectory(_relativeTo) ? getGenericPath(_relativeTo) : getParent(_relativeTo));
```
[the remark this answers is not in the document]
Comment (tomaz82) [reply es-core/src/utils/FileSystemUtil.cpp]: Done

File: es-app/src/Gamelist.cpp [hunk 3]
```diff
@@ -165,7 +165,7 @@ void addFileDataNode(pugi::xml_node& parent, const FileData* file, const char* t
 		//there's something useful in there so we'll keep the node, add the path
 
 		// try and make the path relative if we can so things still work if we change the rom folder location in the future
-		newNode.prepend_child("path").text().set(Utils::FileSystem::createRelativePath(file->getPath(), system->getStartPath(), false).c_str());
+		newNode.prepend_child("path").text().set(Utils::FileSystem::createRelativePath(file->getPath(), system->getStartPath(), false, false).c_str());
```
Comment (aarononeal) [inline es-app/src/Gamelist.cpp:168]: The bool results in a reversal from my PR. Are you sure it is correct?

Comment (tomaz82) [reply es-app/src/Gamelist.cpp]: You're absolutely correct! Fixed it.

**Comment by aarononeal** [comment]
Looks good and looking forward to this optimization being integrated.

**Pull request merged**